horovod/torch/mpi_ops.py

```diff
@@ -0,0 +1,345 @@
+        from root rank.
+    """
+    handle = broadcast_async(tensor, root_rank, name)
+    return synchronize(handle)
+
+
+def broadcast_async_(tensor, root_rank, name=None):
+    """
+    A function that asynchronously broadcasts the input tensor on root rank to the same
+    input tensor on all other Horovod processes. The operation is performed in-place.
+
+    The broadcast operation is keyed by the name. If name is not provided, an incremented
+    auto-generated name is used. The tensor type and shape must be the same on all
+    Horovod processes for a given name. The broadcast will not start until all processes
+    are ready to send and receive the tensor.
+
+    Arguments:
+        tensor: A tensor to broadcast.
+        root_rank: The rank to broadcast the value from.
+        name: A name of the broadcast operation.
+
+    Returns:
+        A handle to the broadcast operation that can be used with `poll()` or
+        `synchronize()`.
+    """
+    return _broadcast_async(tensor, tensor, root_rank, name)
+
+
+def broadcast_(tensor, root_rank, name=None):
+    """
+    A function that broadcasts the input tensor on root rank to the same input tensor
+    on all other Horovod processes. The operation is performed in-place.
+
+    The broadcast operation is keyed by the name. If name is not provided, an incremented
+    auto-generated name is used. The tensor type and shape must be the same on all
+    Horovod processes for a given name. The broadcast will not start until all processes
+    are ready to send and receive the tensor.
+
+    Arguments:
+        tensor: A tensor to broadcast.
+        root_rank: The rank to broadcast the value from.
+        name: A name of the broadcast operation.
+
+    Returns:
+        A tensor of the same shape and type as `tensor`, with the value broadcasted
+        from root rank.
+    """
+    handle = broadcast_async_(tensor, root_rank, name)
+    return synchronize(handle)
+
+
+def poll(handle):
+    """
+    Polls an allreduce, allgather or broadcast handle to determine whether underlying
+    asynchronous operation has completed. After `poll()` returns `True`, `synchronize()`
+    will return without blocking.
+
+    Arguments:
+        handle: A handle returned by an allreduce, allgather or broadcast asynchronous
+                operation.
+
+    Returns:
+        A flag indicating whether the operation has completed.
+    """
+    return mpi_lib.horovod_torch_poll(handle) != 0
+
+
+def synchronize(handle):
+    """
+    Synchronizes an asynchronous allreduce, allgather or broadcast operation until
+    it's completed. Returns the result of the operation.
+
+    Arguments:
+        handle: A handle returned by an allreduce, allgather or broadcast asynchronous
+                operation.
+
+    Returns:
+        An output tensor of the operation.
+    """
+    if handle not in _handle_map:
+        return
+    mpi_lib.horovod_torch_wait_and_clear(handle)
+    _, output = _handle_map[handle]
```

Comment:
You can merge these two lines using pop.

Reply:
Good catch!